narwhals/dependencies.py

```diff
@@ -230,6 +236,14 @@ def is_pyspark_dataframe(df: Any) -> TypeIs[pyspark_sql.DataFrame]:
     )
 
 
+def is_pyspark_connect_dataframe(df: Any) -> TypeIs[PySparkConnectDataFrame]:
+    """Check whether `df` is a PySpark Connect DataFrame without importing PySpark."""
+    return bool(
+        (pyspark_sql := get_pyspark_connect()) is not None
+        and isinstance(df, pyspark_sql.dataframe.DataFrame)
```

Comment:
nitpick (sorry - just because by reading the variable name I initially thought we were importing `pyspark.sql`)

```suggestion
        (pyspark_connect := get_pyspark_connect()) is not None
        and isinstance(df, pyspark_connect.dataframe.DataFrame)
```